rand/rand.go

```diff
@@ -2,14 +2,16 @@ package rand
 
 import (
 	"math/rand"
+	"sync"
 	"time"
 )
 
 /*
  * Top-level convenience functions
  */
 
-var globalRand = rand.New(rand.NewSource(time.Now().UnixNano()).(rand.Source64))
+var globalSource = lockedSource{src: rand.NewSource(time.Now().UnixNano()).(rand.Source64)}
```

Comment:
`rand.Source` isn't thread-safe. So basically applied a similar idea from stdlib and wrapped a Source to be guarded on each method of `rand.Source64`.